sqa/starknet/writer/tables.py

```diff
@@ -104,3 +108,128 @@ def _append_keys(self, keys: list[str]) -> None:
+        self.call_entry_point_selector = Column(pyarrow.string())
+        self.call_entry_point_type = Column(pyarrow.string())
+        self.call_revert_reason = Column(pyarrow.string())
+        self.calldata = Column(pyarrow.list_(pyarrow.string()))
+        self.call_result = Column(pyarrow.list_(pyarrow.string()))
+
+        self.call_events_keys = Column(pyarrow.list_(pyarrow.string()))
+        self.call_events_data = Column(pyarrow.list_(pyarrow.string()))
+        self.call_events_order = Column(pyarrow.list_(pyarrow.int32()))
+
+        self.call_messages_payload = Column(pyarrow.list_(pyarrow.string()))
+        self.call_messages_from_address = Column(pyarrow.list_(pyarrow.string()))
+        self.call_messages_to_address = Column(pyarrow.list_(pyarrow.string()))
+        self.call_messages_order = Column(pyarrow.list_(pyarrow.int32()))
+
+    def append_root(self, tx_trace: WriterTrace):
+        trace_root = tx_trace['trace_root']
+        # NOTE: ensure one and only one call trace presented
+        call_index = 0
+        if 'execute_invocation' in trace_root:
+            call_index += self.append_call(tx_trace, trace_root['execute_invocation'], 'execute_invocation', call_index)
+        if 'constructor_invocation' in trace_root:
+            call_index += self.append_call(tx_trace, trace_root['constructor_invocation'], 'constructor_invocation', call_index)
+        if 'validate_invocation' in trace_root:
+            call_index += self.append_call(tx_trace, trace_root['validate_invocation'], 'validate_invocation', call_index)
+        if 'fee_transfer_invocation' in trace_root:
+            call_index += self.append_call(tx_trace, trace_root['fee_transfer_invocation'], 'fee_transfer_invocation', call_index)
+
+    def append_call(self, tx_trace: WriterTrace, trace_call: TraceCall, invocation_type: str, call_index: int, parent_index: int = -1) -> int:
+        self.block_number.append(tx_trace['block_number'])
+        self.transaction_index.append(tx_trace['transaction_index'])
+        self.trace_type.append(tx_trace['trace_root']['type'])
+        self.invocation_type.append(invocation_type)
+
+        self.parent_index.append(parent_index)
+        self.call_index.append(call_index)
+
+        # NOTE: fields below are empty for reverted calls
+        self.caller_address.append(trace_call.get('caller_address'))
+        self.call_contract_address.append(trace_call.get('contract_address'))
+        self.call_type.append(trace_call.get('call_type'))
+        self.call_class_hash.append(trace_call.get('class_hash'))
+        self.call_entry_point_selector.append(trace_call.get('entry_point_selector'))
+        self.call_entry_point_type.append(trace_call.get('entry_point_type'))
+        self.call_revert_reason.append(trace_call.get('revert_reason'))
+        self.calldata.append(trace_call.get('calldata'))
+        self.call_result.append(trace_call.get('result'))
+
+        # TODO: should call events have own table?
```

Comment:
>if there is no need to filter them now we can store them as a single field which is a list of structs

Then, it will be impossible to add filtering capabilities without breaking changes.